crates/node/src/validation.rs

```diff
@@ -144,14 +147,16 @@ async fn validate_next_block(
                     .get(solution_index)
                     .expect("Failed solution must exist."),
             );
-            let conn = conn_pool.acquire().await.map_err(CriticalError::from)?;
-            tokio::task::spawn_blocking(move || {
-                db::insert_failed_block(&conn, &block_address, &failed_solution)
-                    .map_err(ValidationError::from)?;
-                Ok(false)
-            })
-            .await
-            .map_err(RecoverableError::Join)?
+            let block_address = block_address.clone();
```

Comment:
we might not need to clone here as the block address is *moved* into the closure below, but I could be wrong